packages/lesswrong/server/resolvers/dialogueChecksResolvers.ts

```diff
@@ -1,26 +1,26 @@
 import DialogueChecks from "../../lib/collections/dialogueChecks/collection";
-import { randomId } from "../../lib/random";
 import { augmentFieldsDict } from "../../lib/utils/schemaUtils";
 import { createNotifications } from "../notificationCallbacksHelpers";
 import DialogueChecksRepo from "../repos/DialogueChecksRepo";
 import { defineMutation } from "../utils/serverGraphqlUtil";
 
-async function notifyUsersIfMatchingDialogueChecks (dialogueCheck: DbDialogueCheck) {
-  const match = await getMatchingDialogueCheck(dialogueCheck);
-  if (match) {
-    await createNotifications({
+async function notifyUsersMatchingDialogueChecks (dialogueCheck: DbDialogueCheck, matchingDialogueCheck: DbDialogueCheck, associatedMessage?: DbMessage) {
+  await Promise.all([ 
+    createNotifications({
       userIds: [dialogueCheck.userId],
       notificationType: "newDialogueMatch",
       documentType: "dialogueCheck",
       documentId: dialogueCheck._id,
-    });
-    await createNotifications({
-      userIds: [match.userId],
+      extraData: {associatedMessage}
+    }),
+    createNotifications({
+      userIds: [matchingDialogueCheck.userId],
       notificationType: "newDialogueMatch",
       documentType: "dialogueCheck",
-      documentId: match._id,
-    });
-  }
+      documentId: matchingDialogueCheck._id,
+      extraData: {associatedMessage}
```

Comment:
I don't believe the `extraData: {associatedMessage}` is doing anything, since it's not getting handled by the `newDialogueMatch` notification type.

Reply:
(And I don't even think that it's ever getting passed into `notifyUsersMatchingDialogueChecks`)